Handle searching by multiple names at once

tjozwik opened the pull request:
Resolves #46 

Handles searching by multiple names of the same object (e.g. language) if names are separated by semicolon.
```ruby
# different names of the same language given
2.7.1 :002 > I18nData.language_code('Catalan; Valencian')
 => "CA" 
# names of two different languages given
2.7.1 :003 > I18nData.language_code('Catalan; Polish')
 => nil 
```
Previous behavior:
```ruby
2.7.1 :002 > I18nData.language_code('Catalan; Valencian')
 => nil 
2.7.1 :003 > I18nData.language_code('Catalan; Polish')
 => nil 
```
@grosser

grosser commented:
hmmm seems kinda fishy that it supports arbitrary languages, would it be simpler to only support the known languages with `;` ?

tjozwik commented:
Ok. So after analyzing it, I think that the simpler solution would be changing the return condition from
```ruby
return code if (name.include?(search) && name.split(';').each(&:strip!).include?(search))
```
to:
```ruby
return code if (name.include?(search) && (search_names - name.split(';').each(&:strip!)).empty?)
```
and defining `search_names` before `detect` block as:
```ruby
search_names = search.split(';').each(&:strip!)
```

grosser commented:
fixed in v0.12.0 https://github.com/grosser/i18n_data/commit/353d5d069d58ac7ea0d5107f18f243ce8e6378a3
